RecCalorimeter/src/components/CalibrateBenchmarkMethod.cpp

```diff
@@ -150,126 +144,176 @@ StatusCode CalibrateBenchmarkMethod::execute() {
 
 
 // minimisation function for the benchmark method
-double CalibrateBenchmarkMethod::chiSquareFitBarrel(const Double_t *par) const {
+double CalibrateBenchmarkMethod::chiSquareFitBarrel(const Double_t *parameter) const {
   double fitvalue = 0.;
   // loop over all events, vector of a size of #evts is filled with the energies 
   // ECal calibrated to EM scale, HCal calibrated to HAD scale
-  for(uint i=0; i<m_vecEinECaltotal.size(); i++){
-    double E_generated = m_vecEgenerated.at(i);
-    double E_ECal_total = m_vecEinECaltotal.at(i);
-    double E_ECal_lastLayer = m_vecEinECallastLayer.at(i);
-    double E_HCal_total = m_vecEinHCaltotal.at(i);
-    double E_HCal_firstLayer = m_vecEinHCalfirstLayer.at(i);
-    
-    Double_t E0 = E_ECal_total*par[0] + E_HCal_total*par[1] + par[2]*sqrt(abs(E_ECal_lastLayer*par[0]*E_HCal_firstLayer*par[1])) + par[3]*pow(E_ECal_total*par[0],2);
+  for(uint i=0; i<m_vecTotalEnergyinECal.size(); i++){
+    double generatedEnergy = m_vecGeneratedEnergy.at(i);
+    double totalEnergyInECal = m_vecTotalEnergyinECal.at(i);
+    double totalEnergyInHCal = m_vecTotalEnergyinHCal.at(i);
+    double energyInFirstLayerECal = m_vecEnergyInFirstLayerECal.at(i);
+    double energyInLastLayerECal = m_vecEnergyInLastLayerECal.at(i);
+    double energyInFirstLayerHCal = m_vecEnergyInFirstLayerHCal.at(i);
+
+    Double_t benchmarkEnergy = parameter[0] * totalEnergyInECal +
+                               parameter[1] * totalEnergyInHCal +
+                               parameter[2] * std::sqrt(std::fabs(energyInLastLayerECal * parameter[0] * energyInFirstLayerHCal * parameter[1])) +
+                               parameter[3] * std::pow(totalEnergyInECal * parameter[0], 2) +
+                               parameter[4] * energyInFirstLayerECal +
+                               parameter[5];
+
+    // general formula below gives possibility to fit also parameter[1] to scale E_HCal_total and a contant term  parameter[5] to include residuals 
+    // parameter[0] scales ECal to HAD scale
+    // parameter[1] scales HCal to HAD scale (fixed to 1 if HCal is already calibrated at HAD scale at the input level)
+    // parameter[2] energy losses between ECal and HCal 
+    // parameter[3] corrects for non-compensation of ECal
+    // parameter[4] upstream correction 
+    // parameter[5] residuals 
     
-    fitvalue += pow((E_generated-E0),2)/E_generated; 
+    fitvalue += std::pow((generatedEnergy-benchmarkEnergy),2)/generatedEnergy; 
   }
   return fitvalue;    
 }
 
 
 StatusCode CalibrateBenchmarkMethod::finalize() {
   // the actual minimisation is running here 
-  std::cout << "Running minimisation for " << m_vecEgenerated.size() << " #events! \n";
-  if (m_vecEgenerated.size() != m_vecEinECaltotal.size()){
+  std::cout << "Running minimisation for " << m_vecGeneratedEnergy.size() << " #events! \n";
+  if (m_vecGeneratedEnergy.size() != m_vecTotalEnergyinECal.size()){
     std::cout << "Something's wrong!!" << std::endl; 
```

Comment:
Can you be more specific?

Reply:
Thank you @kjvbrt  for your suggestions, I implemented all of them.